android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -452,9 +362,154 @@ public void onSignalingChange(PeerConnection.SignalingState signalingState) {
         webRTCModule.sendEvent("peerConnectionSignalingStateChanged", params);
     }
 
+    /**
+     * Triggered when a new track is signaled by the remote peer, as a result of
+     * setRemoteDescription.
+     */
     @Override
     public void onAddTrack(final RtpReceiver receiver, final MediaStream[] mediaStreams) {
```

Comment:
Checked the C++ code, this is what the comment says for `OnAddTrack`:

```
  // This is called when a receiver and its track are created.
  // TODO(zhihuang): Make this pure virtual when all subclasses implement it.
  // Note: This is called with both Plan B and Unified Plan semantics. Unified
  // Plan users should prefer OnTrack, OnAddTrack is only called as backwards
  // compatibility (and is called in the exact same situations as OnTrack).
```

So I'd say we drop our `onAddTrack` handler and just use `onTrack`.

Reply:
That seems right. However, there is an issue with regards to fetching the Receiver's associated MediaStreams. It seems that on the C++ side, there is a function to do that referenced here 
```cpp
  // This is called when signaling indicates a transceiver will be receiving
  // media from the remote endpoint. This is fired during a call to
  // SetRemoteDescription. The receiving track can be accessed by:
  // `transceiver->receiver()->track()` and its associated streams by
  // `transceiver->receiver()->streams()`.
  // Note: This will only be called if Unified Plan semantics are specified.
  // This behavior is specified in section 2.2.8.2.5 of the "Set the
  // RTCSessionDescription" algorithm:
  // https://w3c.github.io/webrtc-pc/#set-description
  virtual void OnTrack(
      rtc::scoped_refptr<RtpTransceiverInterface> transceiver) {}

```

which exists inside `rtp_transceiver_interface.h`: 
```cpp

  // The list of streams that `track` is associated with. This is the same as
  // the [[AssociatedRemoteMediaStreams]] internal slot in the spec.
  // https://w3c.github.io/webrtc-pc/#dfn-associatedremotemediastreams
  // TODO(hbos): Make pure virtual as soon as Chromium's mock implements this.
  // TODO(https://crbug.com/webrtc/9480): Remove streams() in favor of
  // stream_ids() as soon as downstream projects are no longer dependent on
  // stream objects.
  virtual std::vector<std::string> stream_ids() const;
  virtual std::vector<rtc::scoped_refptr<MediaStreamInterface>> streams() const;

```
while I couldn't find that on the Android native side. Any ideas?

Reply:
I see it's indeed missing in both Android and iOS. I'd say we make a patch to WebRTC adding them. Thoughts?

Reply:
Wait, this is weird. I checked the API and it doesn't mention any streams property. So AFAIS there is no way to get a reference to the stream from a transceiver object. Am I missing something?

Reply:
Ah right. We need it for the event. So weird.

Reply:
Yep, patching WebRTC seems to be our only option, although for now, it seems to me that onAddTrack still works, so I am not sure if the patch is urgent.